src/services/flatmapQueries.js

```diff
@@ -0,0 +1,339 @@
+    let tooltipData = {
+      destinations: this.destinations, 
+      origins: this.origins,
+      components: this.components,
+      destinationsWithDatasets: this.destinationsWithDatasets,
+      originsWithDatasets: this.originsWithDatasets,
+      componentsWithDatasets: this.componentsWithDatasets,
+      title: eventData.label,
+      featureId: eventData.resource,
+      hyperlinks: eventData.feature.hyperlinks
+    }
+    console.log('tooltipData', tooltipData)
+    return tooltipData
+  }
+
+  createUnfilledTooltipData = function (){
+    let tooltipData = {
+      destinations: [],
+      origins: [],
+      components: [],
+      destinationsWithDatasets: [],
+      originsWithDatasets: [],
+      componentsWithDatasets: [],
+      resource: undefined
+    }
+    return tooltipData
+  }
+
+  getOrganCuries = function(){
+    return new Promise(resolve=> {
+    fetch(`${this.sparcAPI}get-organ-curies/`)
+      .then(response=>response.json())
+      .then(data=>{
+        resolve(data.uberon.array)
+      })
+    })
+  }
+
+  createComponentsLabelList = function(components, lookUp){
+    let labelList = []
+    components.forEach(n=>{
+      labelList.push(this.createLabelFromNeuralNode(n[0]), lookUp)
+      if (n.length === 2){
+        labelList.push(this.createLabelFromNeuralNode(n[1]), lookUp)
+      }
+    })
+    return labelList
+  }
+
+  createLabelLookup = function(uberons) {
+    return new Promise(resolve=> {
+      let uberonMap = {}
+      const data = { sql: this.buildLabelSqlStatement(uberons)}
+      fetch(`${this.flatmapApi}knowledge/query/`, {
+          method: 'POST',
+          headers: {
+            'Content-Type': 'application/json',
+          },
+          body: JSON.stringify(data),
+        })
+        .then(response => response.json())
+        .then(payload => {
+          const entity = payload.keys.indexOf("entity");
+          const label = payload.keys.indexOf("label");
+          if (entity > -1 && label > -1) {
+            payload.values.forEach(pair => {
+              uberonMap[pair[entity]] = pair[label];
+            });
+          }
+        resolve(uberonMap)
+        })
+    })
+  }
+
+  buildConnectivitySqlStatement = function (keastIds) {
+    let sql = 'select knowledge from knowledge where entity in ('
+    if (keastIds.length === 1) {
+      sql += `'${keastIds[0]}')`
+    } else if (keastIds.length > 1) {
+      for (let i in keastIds) {
+        sql += `'${keastIds[i]}'${i >= keastIds.length - 1 ? ')' : ','} `
+      }
+    }
+    return sql
+  }
+  buildLabelSqlStatement = function (uberons) {
+    let sql = 'select entity, label from labels where entity in ('
+    if (uberons.length === 1) {
+      sql += `'${uberons[0]}')`
+    } else if (uberons.length > 1) {
+      for (let i in uberons) {
+        sql += `'${uberons[i]}'${i >= uberons.length - 1 ? ')' : ','} `
+      }
+    }
+    return sql
+  }
+
+  findAllIdsFromConnectivity = function (connectivity) {
```

Comment:
There is a duplicated function at the end of this file and I prefer the style here better.